### PR TITLE
Refactor: Remove _BatchLoader, add tests for DataLoader

### PR DESCRIPTION
**Pull request by rahul-tuli**
Also moved validation of of dataset, batch_size, iter_steps to top of init

### CODE REVIEW

File: tests/sparsezoo/utils/test_data.py [hunk 1]
```diff
@@ -11,204 +11,115 @@
...
+    data = RandomDataset(name=name, typed_shapes=typed_shapes, num_samples=num_samples)
+    yield data
+    del data
 
 
 @pytest.fixture
-def single_input_dataset():
-    data = [np.random.rand(3, 2), np.random.rand(3, 2), np.random.rand(3, 2)]
-    return Dataset(data=data, name="single-input-test-dataset")
+def dataset_b():
```
Comment (kevinaer) [inline tests/sparsezoo/utils/test_data.py:35]: small nit, but can we be more descriptive about the fixture?

File: tests/sparsezoo/utils/test_data.py [hunk 2]
```diff
@@ -11,204 +11,115 @@
...
+from sparsezoo.utils.data import DataLoader, RandomDataset
+
+
+# TESTS for DataLoader
 
 
 @pytest.fixture
-def dummy_dataset():
-    return Dataset(data=[np.random.rand(100, 10)], name="dummy-dataset")
+def dataset():
```
Comment (kevinaer) [inline tests/sparsezoo/utils/test_data.py:27]: Can we add return types to the fixture method definitions?

**Review by kevinaer — CHANGES_REQUESTED** [review]
Nit: can we get return types for the test fixtures?

File: tests/sparsezoo/utils/test_data.py (hunk 1 above)
Comment (rahul-tuli) [reply tests/sparsezoo/utils/test_data.py:35]: Yes sure thing!

File: tests/sparsezoo/utils/test_data.py (hunk 2 above)
Comment (rahul-tuli) [reply tests/sparsezoo/utils/test_data.py]: Done!

**Review by kevinaer — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**